managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -420,7 +420,29 @@ public void closeFailed(ManagedLedgerException exception, Object ctx) {
         });
     }
 
+    @Override
+    public void asyncOpenReadOnlyManagedLedger(String managedLedgerName,
```

Comment:
Can we write this method like this( Saves the new class `AsyncCallbacks .OpenReadOnlyManagedLedgerCallback` )?

```
@Override
public CompletableFuture<ReadOnlyManagedLedgerImpl> asyncOpenReadOnlyManagedLedger(String managedLedgerName,
                          ManagedLedgerConfig config, Object ctx) {
    if (closed) {
        return FutureUtil.failedFuture(new ManagedLedgerException.ManagedLedgerFactoryClosedException());
    }
    ReadOnlyManagedLedgerImpl roManagedLedger = new ReadOnlyManagedLedgerImpl(this,
            bookkeeperFactory
                    .get(new EnsemblePlacementPolicyConfig(config.getBookKeeperEnsemblePlacementPolicyClassName(),
                            config.getBookKeeperEnsemblePlacementPolicyProperties())),
            store, config, scheduledExecutor, managedLedgerName);
    return roManagedLedger.initialize().thenApply(ignore -> roManagedLedger);
}
```

Reply:
@codelipenghui Use public CompletableFuture<ReadOnlyManagedLedgerImpl> asyncOpenReadOnlyManagedLedgercan avoid defining a new inner class.
But other asynchronous methods of ManagedLedgerFactoryImpl.java all use callback, whether it is necessary for us to keep it consistent.

Reply:
It's not required @liangyepianzhou

Reply:
@liangyepianzhou @poorbarcode I think AsyncCallbacks.OpenReadOnlyManagedLedgerCallback is ok. CompletableFuture is not able to pass the context unless you change the returned type.